manager/state/watch/watch.go

```diff
@@ -46,3 +46,8 @@ func (q *Queue) CallbackWatch(matcher events.Matcher) (eventq chan events.Event,
 func (q *Queue) Publish(item events.Event) {
 	q.broadcast.Write(item)
 }
+
+// Close closes the queue and frees the associated resources.
+func (q *Queue) Close() error {
+	return q.broadcast.Close()
```

Comment:
Are you sure? It looks like the `Broadcaster` code shuts down the sinks automatically: https://github.com/docker/go-events/blob/master/broadcast.go#L147

```
                        for _, sink := range b.sinks {
                                if err := sink.Close(); err != nil && err != ErrSinkClosed {
                                        logrus.WithField("events.sink", sink).WithError(err).
                                                Errorf("broadcaster: closing sink failed")
                                }
                        }
```